### DISCUSSION TITLE
Space key unfocusing text box when typing

### DISCUSSION THREAD

**Discussion by Chiarchiaooo** [i3/i3#6020] [Q&A]
As i stated in the title, i have an ubuntu 22.04lts with i3 (first fresh install) that has issues with the space key and every time i press it when typing either disables the keyboard or the textbox/window gets unfocused.
 
I do not know what could be causing this issue, i only know that i had this issue also in kde (with x11) either in flatpak apps or default ones like firefox or the terminal. 

The only customization i've done is set the "action" key (or whatever it's called) to 'Meta' (windows key). 

Thanks in advance for any help

**Comment by KebabLord** [comment]
sounds like you're using a laptop and your space key has an fn function, if that's the case toggle fn off by a keyboard combination or from bios

**Reply by Chiarchiaooo** [reply]
Yeah, An HP one, is there a way to disable the function on my space key without enabling action keys by default again?

Edit: i've just tried with both "fn" key settings and the issue is still there with both.

**Reply by okraits** [reply]
I think in most cases the Fn key is controlled by the bios, so the operating system has no control over it. And i3 only reacts to key events it receives from the X server.
So I guess the only solution is to try to disable the special function on the space key.

**Reply by orestisfl** [reply]
> I think in most cases the Fn key is controlled by the bios, so the operating system has no control over it. And i3 only reacts to key events it receives from the X server. So I guess the only solution is to try to disable the special function on the space key.

An unfocusing text box means the key event is somehow visible

> The only customization i've done is set the "action" key (or whatever it's called) to 'Meta' (windows key).

@Chiarchiaooo does the problem go away if you remove the mapping?

**Reply by Chiarchiaooo** [reply]
@orestisfl I have no idea of how to remove that mapping, tbh.  hahaha

**Reply by KebabLord** [reply]
@Chiarchiaooo it has to be on bios settings, generally on advanced tabs or configuration etc. The setting name might be "action key" or "function key". Just try to disable it or toggle it.
![image](https://github.com/i3/i3/assets/24779257/91230c13-9a05-44dc-8a6b-b7fbef645f4a)

If i'm right, your situation is called Fn lock, and in general it can be disabled from some keyboard combination too, such as holding Fn and pressing Esc etc.

**Reply by Chiarchiaooo** [reply]
@okraits Looking for it,.. you're actually spot on, the fn key is tied to the bios, since it's specific to each manufacturer. 

> An unfocusing text box means the key event is somehow visible
That's what i thought too, but when i pulled out a keyboard tester, it did register the space bar a couple of times, but only when mashing keys to see there were other not functioning.

**Reply by Chiarchiaooo** [reply]
> it has to be on bios settings, generally on advanced tabs or configuration etc. The setting name might be "action key" or "function key". Just try to disable it or toggle it. 

Yeah, mine is an HP Pavilion, so my only option is that button in the bios. Although, as I've said in another response, the issue is still there with both settings.

Should i try purging configs of i3 (and maybe even x11)? Or maybe should re-install ubuntu altogether?

**Reply by orestisfl** [reply]
> Should i try purging configs of i3 (and maybe even x11)? Or maybe should re-install ubuntu altogether?

I would start with a bare minimum X session just with a window manager (i3 with default config, openbox, awesomewm) and see if you can reproduce the problem there.

**Reply by Chiarchiaooo** [reply]
> I would start with a bare minimum X session just with a window manager (i3 with default config, openbox, awesomewm) and see if you can reproduce the problem there.

If possible, i would like to avoid reinstalling ubuntu or remove gnome (which is working fine). I'll try to purge whatever i can and let you know

**Reply by orestisfl** [reply]
> If possible, i would like to avoid reinstalling ubuntu or remove gnome (which is working fine). I'll try to purge whatever i can and let you know

Yes, you don't need to uninstall anything, just start a tty and use startx.

**Reply by Chiarchiaooo** [reply]
> > If possible, i would like to avoid reinstalling ubuntu or remove gnome (which is working fine). I'll try to purge whatever i can and let you know
> 
> Yes, you don't need to uninstall anything, just start a tty and use startx.

It's been two months and ended up just not touching i3 anymore (mostly since I set up a gaming rig and been using that instead of my laptop for the last two months 😂). 

But right now I can use the fn keys without having to press the fn button every time (I envy a friend of mine tho, who can switch it however he pleases)

**Comment by okraits** [comment]
> An unfocusing text box means the key event is somehow visible

But there's no mapping for `space` in the default i3 config which would lead to that effect.

> > The only customization i've done is set the "action" key (or whatever it's called) to 'Meta' (windows key).
> 
> @Chiarchiaooo does the problem go away if you remove the mapping?

I think he means the modifier key.

**Reply by Chiarchiaooo** [reply]
Yeah, the only thing i've done was choose the <meta> key when i3 firstly asked me what i wanted to use as modifier key.

**Comment by okraits** [comment]
> If i'm right, your situation is called Fn lock, and in general it can be disabled from some keyboard combination too, such as holding Fn and pressing Esc etc.

Yes, that's the key combination on my thinkpad as well. If the lock is disabled, then a led under the Fn key is on.

**Reply by Chiarchiaooo** [reply]
No LEDs on mine tho.
